kafka-rest/src/main/java/io/confluent/kafkarest/response/UrlBuilder.java

```diff
@@ -65,6 +65,14 @@ public String build() {
 
     public abstract String getValue();
 
+    public final String encode() {
+      try {
+        return getKey() + "=" + URLEncoder.encode(getValue(), "UTF-8");
```

Comment:
If it was possible to get a `null` value, this will now throw (where it wouldn't before).

AFAICS `AutoValue` is an earlier stop for potential `null` fields, and parsing the URL also probably defaults to an empty `String` instead of a `null`, but I thought I would point it out, in case I'm missing something.

Reply:
How come it was possible to have `null` value before? `AutoValue_UrlBuilder_QueryParameter(key, value)` will throw NPE for `null` key or value, because neither `getKey()` or `getValue()` are marked as `@Nullable`.

Reply:
> If it was possible...

and

> AFAICS AutoValue is an earlier stop for potential null fields

are saying "I don't think it's actually possible" :)

That's also why I ended with

> ...but I thought I would point it out, in case I'm missing something.

Reply:
For some reason my brain skipped the "If" :)